src/other/tables.jl

```diff
@@ -26,7 +26,7 @@ end
 Base.append!(df::DataFrame, x) = append!(df, DataFrame(x))
 
 # This supports the Tables.RowTable type; needed to avoid ambiguities w/ another constructor
-DataFrame(x::Vector{T}) where {T <: NamedTuple} = fromcolumns(Tables.columns(Tables.DataValueUnwrapper(x)))
+DataFrame(x::Vector{<:NamedTuple}) = fromcolumns(Tables.columns(Tables.IteratorWrapper(x)))
```

Comment:
Should this be `AbstractVector`?

Reply:
It can be a custom `AbstractVector` of with, in particular, a most common case a `SubArray`.

Reply:
Guess we can check whether these are supported by Tables and if such move to use `AbstractVector`. I assume this method to avoid ambiguities should work fine with `AbstractVector`.

Reply:
Yes, but my general thinking is that we should make it consistent across the ecosystem, so if something would need to be changed in Tables.jl we should also change it there in sync. `SubArray` is actually quite possible to appear if you would perform subseting of rows using a `view` (which with very wide tables is exactly what you want to do given `NamedTuple` is immutable).